/********************************************************************************
 * Copyright 2023 Wisdom (Fujian) Motor Co.,Ltd.
 * All Rights Reserved.
 * Dept.:Software Department.
 * FILE: io_vars.h
 * Description: This file is automatic generate by HalCanAutoGen.
 * History:
 * 2023-03-19 LinXiaobin
********************************************************************************/

#include "build_control.h"

#ifndef __IO_VARS__
#define __IO_VARS__

/* CAN Message: VCU_BattPowerAllow_buf */
extern uint8_t hld_can0_0xc00fefe_msgReady;                
extern uint8_t hld_can0_0xc00fefe_msgTxFailed;             

/* CAN Message: VCU_MCU_Cmd_buf */
extern uint8_t hld_can0_0xcff6127_msgReady;                
extern uint8_t hld_can0_0xcff6127_msgTxFailed;             

/* CAN Message: VCU_Vehicle_States_buf */
extern uint8_t hld_can0_0xcff6227_msgReady;                
extern uint8_t hld_can0_0xcff6227_msgTxFailed;             

/* CAN Message: uds_tx_buf */
extern uint8_t hld_can0_0x18000701_msgReady;               
extern uint8_t hld_can0_0x18000701_msgTxFailed;            

/* CAN Message: MCU_Errors1_buf */
extern uint8_t hld_can0_0xcfff3f0_received;                
extern uint8_t hld_can0_0xcfff3f0_msgOverRun;              
extern uint8_t hld_can0_0xcfff3f0_timeout;                 
extern uint8_t hld_can0_0xcfff3f0_msgValid;                

/* CAN Message: MCU_Errors2_buf */
extern uint8_t hld_can0_0xcfff4f0_received;                
extern uint8_t hld_can0_0xcfff4f0_msgOverRun;              
extern uint8_t hld_can0_0xcfff4f0_timeout;                 
extern uint8_t hld_can0_0xcfff4f0_msgValid;                

/* CAN Message: MCU_Errors3_buf */
extern uint8_t hld_can0_0xcfff5f0_received;                
extern uint8_t hld_can0_0xcfff5f0_msgOverRun;              
extern uint8_t hld_can0_0xcfff5f0_timeout;                 
extern uint8_t hld_can0_0xcfff5f0_msgValid;                

/* CAN Message: MCU_States1_buf */
extern uint8_t hld_can0_0xcff71f0_received;                
extern uint8_t hld_can0_0xcff71f0_msgOverRun;              
extern uint8_t hld_can0_0xcff71f0_timeout;                 
extern uint8_t hld_can0_0xcff71f0_msgValid;                

/* CAN Message: MCU_States2_buf */
extern uint8_t hld_can0_0xcff72f0_received;                
extern uint8_t hld_can0_0xcff72f0_msgOverRun;              
extern uint8_t hld_can0_0xcff72f0_timeout;                 
extern uint8_t hld_can0_0xcff72f0_msgValid;                

/* CAN Message: MCU_States3_buf */
extern uint8_t hld_can0_0xcff73f0_received;                
extern uint8_t hld_can0_0xcff73f0_msgOverRun;              
extern uint8_t hld_can0_0xcff73f0_timeout;                 
extern uint8_t hld_can0_0xcff73f0_msgValid;                

/* CAN Message: uds_rx_buf */
extern uint8_t hld_can0_0x18000801_received;               
extern uint8_t hld_can0_0x18000801_msgOverRun;             
extern uint8_t hld_can0_0x18000801_timeout;                
extern uint8_t hld_can0_0x18000801_msgValid;               

/* CAN Message: uds_rx_func_buf */
extern uint8_t hld_can0_0x18000901_received;               
extern uint8_t hld_can0_0x18000901_msgOverRun;             
extern uint8_t hld_can0_0x18000901_timeout;                
extern uint8_t hld_can0_0x18000901_msgValid;               

/* CAN Message: CCVS_buf */
extern uint8_t hld_can1_0x18fef100_msgReady;               
extern uint8_t hld_can1_0x18fef100_msgTxFailed;            

/* CAN Message: EVCU3_EACCmd_buf */
extern uint8_t hld_can1_0xcff3e27_msgReady;                
extern uint8_t hld_can1_0xcff3e27_msgTxFailed;             

/* CAN Message: EVCU_MCU1_buf */
extern uint8_t hld_can1_0x18ff3c27_msgReady;               
extern uint8_t hld_can1_0x18ff3c27_msgTxFailed;            

/* CAN Message: EVCU_MCU2_buf */
extern uint8_t hld_can1_0x18ff3d27_msgReady;               
extern uint8_t hld_can1_0x18ff3d27_msgTxFailed;            

/* CAN Message: EVCU_MCU3_buf */
extern uint8_t hld_can1_0x18ff3e27_msgReady;               
extern uint8_t hld_can1_0x18ff3e27_msgTxFailed;            

/* CAN Message: EVCU_Status1_buf */
extern uint8_t hld_can1_0xcff3f27_msgReady;                
extern uint8_t hld_can1_0xcff3f27_msgTxFailed;             

/* CAN Message: EVCU_Status2_buf */
extern uint8_t hld_can1_0x18ff4027_msgReady;               
extern uint8_t hld_can1_0x18ff4027_msgTxFailed;            

/* CAN Message: EVCU_Status3_buf */
extern uint8_t hld_can1_0xcff4127_msgReady;                
extern uint8_t hld_can1_0xcff4127_msgTxFailed;             

/* CAN Message: V2B_VIN_buf */
extern uint8_t hld_can1_0x18e1f3ef_msgReady;               
extern uint8_t hld_can1_0x18e1f3ef_msgTxFailed;            

/* CAN Message: VCU_Battery_Control_buf */
extern uint8_t hld_can1_0x1802f3ef_msgReady;               
extern uint8_t hld_can1_0x1802f3ef_msgTxFailed;            

/* CAN Message: uds_tx_buf */
extern uint8_t hld_can1_0x18102201_msgReady;               
extern uint8_t hld_can1_0x18102201_msgTxFailed;            

/* CAN Message: ABS1_VSO_buf */
extern uint8_t hld_can1_0x180b100b_received;               
extern uint8_t hld_can1_0x180b100b_msgOverRun;             
extern uint8_t hld_can1_0x180b100b_timeout;                
extern uint8_t hld_can1_0x180b100b_msgValid;               

/* CAN Message: B2V_CurrentLimit_buf */
extern uint8_t hld_can1_0x18c4fef3_received;               
extern uint8_t hld_can1_0x18c4fef3_msgOverRun;             
extern uint8_t hld_can1_0x18c4fef3_timeout;                
extern uint8_t hld_can1_0x18c4fef3_msgValid;               

/* CAN Message: B2V_RQ_buf */
extern uint8_t hld_can1_0x1801eff3_received;               
extern uint8_t hld_can1_0x1801eff3_msgOverRun;             
extern uint8_t hld_can1_0x1801eff3_timeout;                
extern uint8_t hld_can1_0x1801eff3_msgValid;               

/* CAN Message: B2V_ST1_buf */
extern uint8_t hld_can1_0x1883eff3_received;               
extern uint8_t hld_can1_0x1883eff3_msgOverRun;             
extern uint8_t hld_can1_0x1883eff3_timeout;                
extern uint8_t hld_can1_0x1883eff3_msgValid;               

/* CAN Message: B2V_ST2_buf */
extern uint8_t hld_can1_0x1884eff3_received;               
extern uint8_t hld_can1_0x1884eff3_msgOverRun;             
extern uint8_t hld_can1_0x1884eff3_timeout;                
extern uint8_t hld_can1_0x1884eff3_msgValid;               

/* CAN Message: DCAC_HV_Oilpump_Status1_buf */
extern uint8_t hld_can1_0x18f00828_received;               
extern uint8_t hld_can1_0x18f00828_msgOverRun;             
extern uint8_t hld_can1_0x18f00828_timeout;                
extern uint8_t hld_can1_0x18f00828_msgValid;               

/* CAN Message: DCAC_HV_Oilpump_Status2_buf */
extern uint8_t hld_can1_0x18f00d28_received;               
extern uint8_t hld_can1_0x18f00d28_msgOverRun;             
extern uint8_t hld_can1_0x18f00d28_timeout;                
extern uint8_t hld_can1_0x18f00d28_msgValid;               

/* CAN Message: DCDC1_Status1_buf */
extern uint8_t hld_can1_0x18f00a28_received;               
extern uint8_t hld_can1_0x18f00a28_msgOverRun;             
extern uint8_t hld_can1_0x18f00a28_timeout;                
extern uint8_t hld_can1_0x18f00a28_msgValid;               

/* CAN Message: DCDC1_Status2_buf */
extern uint8_t hld_can1_0x18f00b28_received;               
extern uint8_t hld_can1_0x18f00b28_msgOverRun;             
extern uint8_t hld_can1_0x18f00b28_timeout;                
extern uint8_t hld_can1_0x18f00b28_msgValid;               

/* CAN Message: DS_AP1_buf */
extern uint8_t hld_can1_0x18fc0011_received;               
extern uint8_t hld_can1_0x18fc0011_msgOverRun;             
extern uint8_t hld_can1_0x18fc0011_timeout;                
extern uint8_t hld_can1_0x18fc0011_msgValid;               

/* CAN Message: EPB_Status_buf */
extern uint8_t hld_can1_0x18fe1264_received;               
extern uint8_t hld_can1_0x18fe1264_msgOverRun;             
extern uint8_t hld_can1_0x18fe1264_timeout;                
extern uint8_t hld_can1_0x18fe1264_msgValid;               

/* CAN Message: IC_Status1_buf */
extern uint8_t hld_can1_0x18ffe342_received;               
extern uint8_t hld_can1_0x18ffe342_msgOverRun;             
extern uint8_t hld_can1_0x18ffe342_timeout;                
extern uint8_t hld_can1_0x18ffe342_msgValid;               

/* CAN Message: IC_Status5_buf */
extern uint8_t hld_can1_0x18ffe742_received;               
extern uint8_t hld_can1_0x18ffe742_msgOverRun;             
extern uint8_t hld_can1_0x18ffe742_timeout;                
extern uint8_t hld_can1_0x18ffe742_msgValid;               

/* CAN Message: LVOP_Status1_buf */
extern uint8_t hld_can1_0xcf603a1_received;                
extern uint8_t hld_can1_0xcf603a1_msgOverRun;              
extern uint8_t hld_can1_0xcf603a1_timeout;                 
extern uint8_t hld_can1_0xcf603a1_msgValid;                

/* CAN Message: PDU_Status1_buf */
extern uint8_t hld_can1_0xc00fefe_received;                
extern uint8_t hld_can1_0xc00fefe_msgOverRun;              
extern uint8_t hld_can1_0xc00fefe_timeout;                 
extern uint8_t hld_can1_0xc00fefe_msgValid;                

/* CAN Message: uds_rx_buf */
extern uint8_t hld_can1_0x18102301_received;               
extern uint8_t hld_can1_0x18102301_msgOverRun;             
extern uint8_t hld_can1_0x18102301_timeout;                
extern uint8_t hld_can1_0x18102301_msgValid;               

/* CAN Message: uds_rx_func_buf */
extern uint8_t hld_can1_0x18102401_received;               
extern uint8_t hld_can1_0x18102401_msgOverRun;             
extern uint8_t hld_can1_0x18102401_timeout;                
extern uint8_t hld_can1_0x18102401_msgValid;               

/* CAN Message: LVOP_Control_buf */
extern uint8_t hld_can2_0x18f10527_msgReady;               
extern uint8_t hld_can2_0x18f10527_msgTxFailed;            

/* CAN Message: ccp_daq_0_buf */
extern uint8_t hld_can2_0x219_msgReady;                    
extern uint8_t hld_can2_0x219_msgTxFailed;                 

/* CAN Message: ccp_daq_1_buf */
extern uint8_t hld_can2_0x220_msgReady;                    
extern uint8_t hld_can2_0x220_msgTxFailed;                 

/* CAN Message: ccp_daq_2_buf */
extern uint8_t hld_can2_0x221_msgReady;                    
extern uint8_t hld_can2_0x221_msgTxFailed;                 

/* CAN Message: ccp_daq_3_buf */
extern uint8_t hld_can2_0x222_msgReady;                    
extern uint8_t hld_can2_0x222_msgTxFailed;                 

/* CAN Message: ccp_daq_4_buf */
extern uint8_t hld_can2_0x223_msgReady;                    
extern uint8_t hld_can2_0x223_msgTxFailed;                 

/* CAN Message: ccp_daq_5_buf */
extern uint8_t hld_can2_0x224_msgReady;                    
extern uint8_t hld_can2_0x224_msgTxFailed;                 

/* CAN Message: ccp_daq_6_buf */
extern uint8_t hld_can2_0x225_msgReady;                    
extern uint8_t hld_can2_0x225_msgTxFailed;                 

/* CAN Message: ccp_daq_7_buf */
extern uint8_t hld_can2_0x226_msgReady;                    
extern uint8_t hld_can2_0x226_msgTxFailed;                 

/* CAN Message: ccp_daq_8_buf */
extern uint8_t hld_can2_0x227_msgReady;                    
extern uint8_t hld_can2_0x227_msgTxFailed;                 

/* CAN Message: ccp_rx_buf */
extern uint8_t hld_can2_0x729_msgReady;                    
extern uint8_t hld_can2_0x729_msgTxFailed;                 

/* CAN Message: uds_tx_buf */
extern uint8_t hld_can2_0x18daf110_msgReady;               
extern uint8_t hld_can2_0x18daf110_msgTxFailed;            

/* CAN Message: vin_resp_buf */
extern uint8_t hld_can2_0x18daf103_msgReady;               
extern uint8_t hld_can2_0x18daf103_msgTxFailed;            

/* CAN Message: LVOP_Status1_buf */
extern uint8_t hld_can2_0xcf603a1_received;                
extern uint8_t hld_can2_0xcf603a1_msgOverRun;              
extern uint8_t hld_can2_0xcf603a1_timeout;                 
extern uint8_t hld_can2_0xcf603a1_msgValid;                

/* CAN Message: ccp_tx_buf */
extern uint8_t hld_can2_0x739_received;                    
extern uint8_t hld_can2_0x739_msgOverRun;                  
extern uint8_t hld_can2_0x739_timeout;                     
extern uint8_t hld_can2_0x739_msgValid;                    

/* CAN Message: uds_rx_buf */
extern uint8_t hld_can2_0x18da10f1_received;               
extern uint8_t hld_can2_0x18da10f1_msgOverRun;             
extern uint8_t hld_can2_0x18da10f1_timeout;                
extern uint8_t hld_can2_0x18da10f1_msgValid;               

/* CAN Message: uds_rx_func_buf */
extern uint8_t hld_can2_0x18db33f1_received;               
extern uint8_t hld_can2_0x18db33f1_msgOverRun;             
extern uint8_t hld_can2_0x18db33f1_timeout;                
extern uint8_t hld_can2_0x18db33f1_msgValid;               

extern uint8_t VCU_BattPowerAllow_buf[8];
extern uint8_t VCU_MCU_Cmd_buf[8];
extern uint8_t VCU_Vehicle_States_buf[8];
extern uint8_t uds_tx[8];
extern uint8_t MCU_Errors1_buf[8];
extern uint8_t MCU_Errors2_buf[8];
extern uint8_t MCU_Errors3_buf[8];
extern uint8_t MCU_States1_buf[8];
extern uint8_t MCU_States2_buf[8];
extern uint8_t MCU_States3_buf[8];
extern uint8_t uds_rx[8];
extern uint8_t uds_rx_func[8];
extern uint8_t CCVS_buf[8];
extern uint8_t EVCU3_EACCmd_buf[8];
extern uint8_t EVCU_MCU1_buf[8];
extern uint8_t EVCU_MCU2_buf[8];
extern uint8_t EVCU_MCU3_buf[8];
extern uint8_t EVCU_Status1_buf[8];
extern uint8_t EVCU_Status2_buf[8];
extern uint8_t EVCU_Status3_buf[8];
extern uint8_t V2B_VIN_buf[8];
extern uint8_t VCU_Battery_Control_buf[8];
extern uint8_t uds_tx[8];
extern uint8_t ABS1_VSO_buf[8];
extern uint8_t B2V_CurrentLimit_buf[8];
extern uint8_t B2V_RQ_buf[8];
extern uint8_t B2V_ST1_buf[8];
extern uint8_t B2V_ST2_buf[8];
extern uint8_t DCAC_HV_Oilpump_Status1_buf[8];
extern uint8_t DCAC_HV_Oilpump_Status2_buf[8];
extern uint8_t DCDC1_Status1_buf[8];
extern uint8_t DCDC1_Status2_buf[8];
extern uint8_t DS_AP1_buf[8];
extern uint8_t EPB_Status_buf[8];
extern uint8_t IC_Status1_buf[8];
extern uint8_t IC_Status5_buf[8];
extern uint8_t LVOP_Status1_buf[8];
extern uint8_t PDU_Status1_buf[8];
extern uint8_t uds_rx[8];
extern uint8_t uds_rx_func[8];
extern uint8_t LVOP_Control_buf[8];
extern uint8_t ccp_daq_0[8];
extern uint8_t ccp_daq_1[8];
extern uint8_t ccp_daq_2[8];
extern uint8_t ccp_daq_3[8];
extern uint8_t ccp_daq_4[8];
extern uint8_t ccp_daq_5[8];
extern uint8_t ccp_daq_6[8];
extern uint8_t ccp_daq_7[8];
extern uint8_t ccp_daq_8[8];
extern uint8_t ccp_rx[8];
extern uint8_t uds_tx[8];
extern uint8_t vin_resp_buf[8];
extern uint8_t LVOP_Status1_buf[8];
extern uint8_t ccp_tx[8];
extern uint8_t uds_rx[8];
extern uint8_t uds_rx_func[8];

/* VCU_BattPowerAllow Message Signals */
extern uint16_t pfc_PackCurrent;
extern uint16_t pfc_PackInsideVoltage;
extern uint16_t pfc_AllowMaxDischargeCurr;
extern uint16_t pfc_AllowMaxChargeCurr;

/* VCU_MCU_Cmd Message Signals */
extern uint8_t  pfc_SystemReady;
extern uint8_t  pfc_ParkingSystemState;
extern uint8_t  pfc_FootBrakeState;
extern uint8_t  pfc_MCUEnable;
extern uint8_t  pfc_MCUControlMode;
extern uint8_t  pfc_MCUDirection;
extern uint16_t pfc_DemandTorque;
extern uint16_t pfc_DemandSpeed;
extern uint8_t  pfc_VCULife1ToMCU;

/* VCU_Vehicle_States Message Signals */
extern uint8_t  pfc_AccePedalPosition;
extern uint8_t  pfc_BrakePedalPosition;
extern uint8_t  pfc_AntiRollBackSwitch;
extern uint8_t  pfc_MotorSpeedLimitSwitch;
extern uint8_t  pfc_SlideSwitch;
extern uint8_t  pfc_VCUCrawlOff;
extern uint16_t pfc_VehicleSpeedLimitMotorSpeed;
extern uint8_t  pfc_VCULife2ToMCU;

/* uds_tx Message Signals */

/* MCU_Errors1 Message Signals */
extern uint8_t  pfc_MCUUnderVoltLv4;
extern uint8_t  pfc_MotorOverTempLv4;
extern uint8_t  pfc_MCUOverTempLv4;
extern uint8_t  pfc_MCUDCOverVoltLv4;
extern uint8_t  pfc_MotorUPhaseOverCurrLv4;
extern uint8_t  pfc_MotorVPhaseOverCurrLv4;
extern uint8_t  pfc_MotorWPhaseOverCurrLv4;
extern uint8_t  pfc_MotorOverSpeedLv4;
extern uint8_t  pfc_MotorInterPhaseShortCircLv4;
extern uint8_t  pfc_MCUDCShortCircGNDLv4;
extern uint8_t  pfc_MCUPhaseShortCircGNDLv4;
extern uint8_t  pfc_VehicleCANFault;
extern uint8_t  pfc_MotorCANFault;
extern uint8_t  pfc_CANTimeoutFault;
extern uint8_t  pfc_CANCRCCheckFault;
extern uint8_t  pfc_CANOfflineFault;
extern uint8_t  pfc_MCUTempSensorFault;
extern uint8_t  pfc_MotorTempSensorFault;
extern uint8_t  pfc_MCUUPhaseCurrSensorFault;
extern uint8_t  pfc_MCUVPhaseCurrSensorFault;
extern uint8_t  pfc_MCUWPhaseCurrSensorFault;
extern uint8_t  pfc_MCUUPhaseLossFault;
extern uint8_t  pfc_MCUVPhaseLossFault;
extern uint8_t  pfc_MCUWPhaseLossFault;
extern uint8_t  pfc_LowVoltBatteryFault;
extern uint8_t  pfc_MotorRDCFault;
extern uint8_t  pfc_MCUDCVoltSensorFault;
extern uint8_t  pfc_MCUIBGTFault;
extern uint8_t  pfc_MotorSpeedSensorFault;
extern uint8_t  pfc_MCUEEPROMFault;
extern uint8_t  pfc_MCUActiveDisChargeFault;

/* MCU_Errors2 Message Signals */
extern uint8_t  pfc_MCUUnderVoltLvl3;
extern uint8_t  pfc_MotorOverTempLv3;
extern uint8_t  pfc_MCUOverTempLv3;
extern uint8_t  pfc_MCUDCOverVoltLv3;
extern uint8_t  pfc_MotorUPhaseOverCurrLv3;
extern uint8_t  pfc_MotorVPhaseOverCurrLv3;
extern uint8_t  pfc_MotorWPhaseOverCurrLv3;
extern uint8_t  pfc_MotorOverSpeedLv3;
extern uint8_t  pfc_MCUUnderVoltLv2;
extern uint8_t  pfc_MotorOverTempLv2;
extern uint8_t  pfc_MCUOverTempLv2;
extern uint8_t  pfc_MCUDCOverVoltLv2;
extern uint8_t  pfc_MotorUPhaseOverCurrLv2;
extern uint8_t  pfc_MotorVPhaseOverCurrLv2;
extern uint8_t  pfc_MotorWPhaseOverCurrLv2;
extern uint8_t  pfc_MotorOverSpeedLevel2;

/* MCU_Errors3 Message Signals */
extern uint8_t  pfc_MCUUnderVoltLv1;
extern uint8_t  pfc_MotorOverTempLvl1;
extern uint8_t  pfc_MCUOverTempLv1;
extern uint8_t  pfc_MCUDCOverVoltLv1;
extern uint8_t  pfc_MotorUPhaseOverCurrLv1;
extern uint8_t  pfc_MotorVPhaseOverCurrLv1;
extern uint8_t  pfc_MotorWPhaseOverCurrLv1;
extern uint8_t  pfc_MotorOverSpeedLv1;

/* MCU_States1 Message Signals */
extern uint16_t pfc_MCUDCCurrent;
extern uint16_t pfc_MCUDCVoltage;
extern uint16_t pfc_MCUACCurrent;
extern uint8_t  pfc_MotorTemperature;
extern uint8_t  pfc_MCUTemperature;

/* MCU_States2 Message Signals */
extern uint8_t  pfc_MCUPreChargeAllow;
extern uint8_t  pfc_MCUActiveDisChargeFeedback;
extern uint8_t  pfc_MCUIGBTEnableFeedback;
extern uint8_t  pfc_MCUFunctionsIndicate;
extern uint8_t  pfc_MCUWorkMode;
extern uint8_t  pfc_MCUFaultGrade;
extern uint8_t  pfc_MCUFaultCode;
extern uint16_t pfc_MCUActualMotorSpeed;
extern uint16_t pfc_MCUActualMotorTorque;
extern uint8_t  pfc_VCULife3ToMCU;

/* MCU_States3 Message Signals */
extern uint16_t pfc_MCUMaxDriveTorqueAvailable;
extern uint16_t pfc_MCUMaxBrakeTorqueAvailable;

/* uds_rx Message Signals */

/* uds_rx_func Message Signals */

/* CCVS Message Signals */
extern uint16_t pfc_WheelBasedVehicleVelocity;
extern uint8_t  pfc_HandBrake;
extern uint8_t  pfc_BrakeSwitch;

/* EVCU3_EACCmd Message Signals */
extern uint8_t  pfc_PDUAirConditionRelayCmd;
extern uint8_t  pfc_PDUOilPumpRelayCmd;
extern uint8_t  pfc_PDUDefrostingRelayCmd;
extern uint8_t  pfc_PDUSUDCDCRelayCmd;
extern uint8_t  pfc_PDUSDDCDCEnable;
extern uint8_t  pfc_PDUAirPumpRelayCmd;
extern uint8_t  pfc_PDUMainPositiveRelayCmd;
extern uint8_t  pfc_PDUPreChargeRelayCmd;
extern uint8_t  pfc_PDUOilPumpEnable;
extern uint8_t  pfc_PDUPumpPreChargeRelayCmd;
extern uint8_t  pfc_PDUAirPumpRelayCmd;
extern uint8_t  pfc_PDUSDDCDCRelayCmd;
extern uint8_t  pfc_PDUHeaterRelayCmd;
extern uint8_t  pfc_PDUPTCRelayCmd;

/* EVCU_MCU1 Message Signals */
extern uint16_t pfc_MCUDCCurrent;
extern uint16_t pfc_MCUDCVoltage;
extern uint16_t pfc_MCUACCurrent;

/* EVCU_MCU2 Message Signals */
extern uint16_t pfc_Motor1Speed;
extern uint8_t  pfc_Motor1Temperature;
extern uint8_t  pfc_MCU1Temperature;
extern uint16_t pfc_Motor1Torque;
extern uint8_t  pfc_Motor1TempAlarm;
extern uint8_t  pfc_MCU1TempAlarm;
extern uint8_t  pfc_MotorNumber;
extern uint8_t  pfc_MotorOrderNumber;
extern uint8_t  pfc_Motor1State;
extern uint8_t  pfc_Motor1FunctionIndicate;

/* EVCU_MCU3 Message Signals */
extern uint8_t  pfc_MCUFaultCode;

/* EVCU_Status1 Message Signals */
extern uint8_t  pfc_AccePedalState;
extern uint8_t  pfc_RegenTorqueState;
extern uint8_t  pfc_HandBrake;
extern uint8_t  pfc_FootBrake;
extern uint8_t  pfc_AccePedalPosition;
extern uint8_t  pfc_BrakePedalPosition;
extern uint8_t  pfc_DriveMode;
extern uint8_t  pfc_ActualGearDNRState;
extern uint8_t  pfc_ChargingMode;
extern uint8_t  pfc_EVMode;
extern uint8_t  pfc_ReadyState;
extern uint8_t  pfc_MCUFaultGrade;
extern uint8_t  pfc_DSFault;
extern uint8_t  pfc_MCUState;
extern uint8_t  pfc_VehicleInterLockState;

/* EVCU_Status2 Message Signals */
extern uint8_t  pfc_VehicleFaultGrade;
extern uint16_t pfc_VehicleFaultCode;

/* EVCU_Status3 Message Signals */
extern uint8_t  pfc_VCUWakeUpState;
extern uint8_t  pfc_KeyStartRequest;
extern uint8_t  pfc_MotorCoolFan;
extern uint8_t  pfc_VCULife;

/* V2B_VIN Message Signals */
extern uint8_t  pfc_SerialNumberOfFrameVIN;
extern uint8_t  pfc_VehicleVINCharacter1815;
extern uint8_t  pfc_VehicleVINCharacter2916;
extern uint8_t  pfc_VehicleVINCharacter31017;
extern uint8_t  pfc_VehicleVINCharacter411;
extern uint8_t  pfc_VehicleVINCharacter512;
extern uint8_t  pfc_VehicleVINCharacter613;
extern uint8_t  pfc_VehicleVINCharacter714;

/* VCU_Battery_Control Message Signals */
extern uint8_t  pfc_VCULife;
extern uint8_t  pfc_BMSMainNegaRelayCmd;
extern uint8_t  pfc_BMSDCACRelayCmd;
extern uint8_t  pfc_BMSSUDCDCRelayCmd;
extern uint8_t  pfc_BMSSDDCDCRelayCmd;
extern uint8_t  pfc_VehicleVelocity;
extern uint8_t  pfc_PDUMainPosiRelayState;
extern uint8_t  pfc_PDUMainPosiRelayFault;
extern uint8_t  pfc_BMSMainNegaRelayState;
extern uint8_t  pfc_BMSMainNegaRelayFault;
extern uint8_t  pfc_PDUChargePosiRelayState;
extern uint8_t  pfc_PDUChargePosiRelayFault;
extern uint8_t  pfc_PDUHeaterPosiRelayState;
extern uint8_t  pfc_PDUHeaterPosiRelayFault;
extern uint8_t  pfc_PDUHeaterNegaRelayState;
extern uint8_t  pfc_PDUHeaterNegaRelayFault;
extern uint8_t  pfc_PDUSDDCDCRelayState;
extern uint8_t  pfc_PDUSDDCDCRelayFault;
extern uint8_t  pfc_PDUOilPumpRelayState;
extern uint8_t  pfc_PDUOilPumpRelayFault;
extern uint8_t  pfc_PDUAirPumpRelayState;
extern uint8_t  pfc_PDUAirPumpRelayFault;
extern uint8_t  pfc_PDUPreChargeRelayState;
extern uint8_t  pfc_PDUPreChargeRelayFault;
extern uint8_t  pfc_PDUDefrostingRelayState;
extern uint8_t  pfc_PDUDefrostingRelayFault;

/* uds_tx Message Signals */

/* ABS1_VSO Message Signals */
extern uint8_t  pfc_ABSActive;
extern uint8_t  pfc_ABSEBDActive;
extern uint8_t  pfc_ABSEBDFault;
extern uint8_t  pfc_ABSFault;
extern uint16_t pfc_ABSVehicleVelocity;
extern uint8_t  pfc_ABSVehicleVelocityValid;
extern uint8_t  pfc_ABSNOEVFlag;
extern uint8_t  pfc_ABSLife1;
extern uint8_t  pfc_ABSChecksum1;

/* B2V_CurrentLimit Message Signals */
extern uint16_t pfc_BMSAllowDisChargeCurr;
extern uint16_t pfc_BMSAllowChargeCurr;
extern uint16_t pfc_BMSAllowSustainChargeCurr;
extern uint16_t pfc_BMSAllowSustainDisChargeCurr;

/* B2V_RQ Message Signals */
extern uint8_t  pfc_BMSRequestVIN;

/* B2V_ST1 Message Signals */
extern uint8_t  pfc_BMSMainNegeRelayState;
extern uint8_t  pfc_BMSChargePosiRelay1State;
extern uint8_t  pfc_BMSChargeNegaRelay1State;
extern uint8_t  pfc_BMSChargePosiRelay2State;
extern uint8_t  pfc_BMSChargeNegaRelay2State;
extern uint8_t  pfc_BMSHeaterPosiRelayState;
extern uint8_t  pfc_BMSHeaterNegaRelayState;
extern uint8_t  pfc_BMSCurrentHighVoltageState;
extern uint8_t  pfc_BMSChargeGunConnectState;
extern uint8_t  pfc_BMSChargeState;

/* B2V_ST2 Message Signals */
extern uint8_t  pfc_BMSSOC;
extern uint8_t  pfc_BMSSOH;
extern uint16_t pfc_BMSCurrent;
extern uint16_t pfc_BMSVoltage;
extern uint8_t  pfc_BMSFaultCode;
extern uint8_t  pfc_BMSHighestFaultGrade;
extern uint8_t  pfc_BMSRequestHighVoltPowerOff;
extern uint8_t  pfc_BMSLife;

/* DCAC_HV_Oilpump_Status1 Message Signals */
extern uint8_t  pfc_OilPumpFaultGrade;
extern uint8_t  pfc_OilPumpLife;
extern uint8_t  pfc_OilPumpFaultCode;

/* DCAC_HV_Oilpump_Status2 Message Signals */
extern uint16_t pfc_OilPumpSpeed;
extern uint16_t pfc_OilPumpInputVoltage;
extern uint8_t  pfc_OilPumpCurrent;
extern uint8_t  pfc_OilPumpInputMCUTemp;
extern uint8_t  pfc_OilPumpInputMotorTemp;
extern uint8_t  pfc_OilPumpPreChargeState;
extern uint8_t  pfc_OilPumpRunningState;

/* DCDC1_Status1 Message Signals */
extern uint8_t  pfc_SDDCDCFaultGrade;
extern uint8_t  pfc_SDDCDCLife;
extern uint8_t  pfc_SDDCDCFaultCode;

/* DCDC1_Status2 Message Signals */
extern uint16_t pfc_SDDCDCInputVoltage;
extern uint8_t  pfc_SDDCDCInputCurrent;
extern uint8_t  pfc_SDDCDCOutputCurrent;
extern uint8_t  pfc_SDDCDCTemp;
extern uint8_t  pfc_SDDCDCOutputVoltage;
extern uint8_t  pfc_SDDCDCPreChargeState;
extern uint8_t  pfc_SDDCDCRunningState;
extern uint8_t  pfc_SDDCDCFaultState;
extern uint8_t  pfc_SDDCDCTempFault;

/* DS_AP1 Message Signals */
extern uint8_t  pfc_DSSelfCheckState;
extern uint8_t  pfc_DSGearD;
extern uint8_t  pfc_DSGearN;
extern uint8_t  pfc_DSGearR;
extern uint8_t  pfc_DSLife;

/* EPB_Status Message Signals */
extern uint8_t  pfc_EPBParkLampState;
extern uint8_t  pfc_EPBSystemFault;
extern uint8_t  pfc_EPBEmgBrakeLampRequest;
extern uint8_t  pfc_EPBWorkMode;
extern uint8_t  pfc_EPBSwitchState;
extern uint8_t  pfc_EPBSystemState;
extern uint8_t  pfc_EPBFaultCode;
extern uint8_t  pfc_EPBLife;
extern uint8_t  pfc_EPBCheckSum;

/* IC_Status1 Message Signals */
extern uint8_t  pfc_DriverDoorState;
extern uint8_t  pfc_RearHatchState;
extern uint8_t  pfc_AccePedalInterLockState;

/* IC_Status5 Message Signals */
extern uint8_t  pfc_MotorInputCoolingWaterTemp;

/* LVOP_Status1 Message Signals */
extern uint8_t  pfc_LVOilPumpRunningState;
extern uint8_t  pfc_LVOilPumpInputVoltage;
extern uint8_t  pfc_LVOilPumpCurrent;
extern uint8_t  pfc_LVOilPumpMotorSpeed;
extern uint8_t  pfc_LVOilPumpLife;
extern uint8_t  pfc_LVOilPumpMCUTemp;
extern uint8_t  pfc_LVOilPumpFaultCode;
extern uint8_t  pfc_LVOilPumpFaultGrade;

/* PDU_Status1 Message Signals */
extern uint8_t  pfc_PDUSDDCDCRelayState;
extern uint8_t  pfc_PDUSDDCDCRelayFault;
extern uint8_t  pfc_PDUAirPumpRelayState;
extern uint8_t  pfc_PDUAirPumpRelayFault;
extern uint8_t  pfc_PDUHeaterRelayState;
extern uint8_t  pfc_PDUHeaterRelayFault;
extern uint8_t  pfc_PDUPTCRelayState;
extern uint8_t  pfc_PDUPTCRelayFault;
extern uint8_t  pfc_PDUPumpPreChargeRelayState;
extern uint8_t  pfc_PDUPumpPreChargeRelayFault;
extern uint8_t  pfc_PDUSUDCDCRelayState;
extern uint8_t  pfc_PDUSUDCDCRelayFault;
extern uint8_t  pfc_PDUAirConditionRelayState;
extern uint8_t  pfc_PDUAirConditionRelayFault;
extern uint8_t  pfc_PDUDefrostingRelayState;
extern uint8_t  pfc_PDUDefrostingRelayFault;
extern uint8_t  pfc_PDUOilPumpRelayState;
extern uint8_t  pfc_PDUOilPumpRelayFault;
extern uint8_t  pfc_PDUMainPosiRelayState;
extern uint8_t  pfc_PDUMainPosiRelayFault;
extern uint8_t  pfc_PDUPreChargeRelayState;
extern uint8_t  pfc_PDUPreChargeRelayFault;

/* uds_rx Message Signals */

/* uds_rx_func Message Signals */

/* LVOP_Control Message Signals */
extern uint8_t  pfc_LVOilPumpEnable;
extern uint8_t  pfc_VehicleVelocity;
extern uint8_t  pfc_LVOilPumpLife;

/* ccp_daq_0 Message Signals */

/* ccp_daq_1 Message Signals */

/* ccp_daq_2 Message Signals */

/* ccp_daq_3 Message Signals */

/* ccp_daq_4 Message Signals */

/* ccp_daq_5 Message Signals */

/* ccp_daq_6 Message Signals */

/* ccp_daq_7 Message Signals */

/* ccp_daq_8 Message Signals */

/* ccp_rx Message Signals */

/* uds_tx Message Signals */

/* vin_resp Message Signals */

/* LVOP_Status1 Message Signals */
extern uint8_t  pfc_LVOilPumpRunningState;
extern uint8_t  pfc_LVOilPumpInputVoltage;
extern uint8_t  pfc_LVOilPumpCurrent;
extern uint8_t  pfc_LVOilPumpMotorSpeed;
extern uint8_t  pfc_LVOilPumpLife;
extern uint8_t  pfc_LVOilPumpMCUTemp;
extern uint8_t  pfc_LVOilPumpFaultCode;
extern uint8_t  pfc_LVOilPumpFaultGrade;

/* ccp_tx Message Signals */

/* uds_rx Message Signals */

/* uds_rx_func Message Signals */


#endif /*End of __IO_VARS__*/
/*-------------------------------------EOF--------------------------------------*/

